packages/plugin-infra/src/types.ts

```diff
@@ -137,8 +138,8 @@ export namespace Plugin.Shared {
         web3?: Web3EnableRequirement
     }
     export interface Web3EnableRequirement {
-        /** This flag indicates the plugin entry in the composition entry should be hidden if the current chain is invalid. */
-        compositionEntryRequiresChainIDValid?: boolean
+        /** Plugin can declare what chain it supports. When the current chain is not supported, the composition entry will be hidden. */
+        requiredChains?: ChainId[]
```

Comment:
This name is misleading. It only hides the entrance for some cases but it didn't reflected that in the name.